### ISSUE TITLE
[WAFv2] resource/wafv2_rule_group: recreated on any changes to the rules 

### ISSUE THREAD

**Issue by anGie44** [terraform-providers/terraform-provider-aws#14071]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

Relates #13936

### Terraform Version
```
Terraform v0.12.26
provider.aws v2.67.0
```

### Affected Resource(s)

```
aws_wafv2_rule_group
```

### Terraform Configuration Files

```hcl
resource "aws_wafv2_rule_group" "example" {
  capacity    = 50
  name        = "tf-acc-test-123"
  description = "tf-acc-test-123"
  scope       = "REGIONAL"

  rule {
    name     = "rule-2"
    priority = 10

    action {
      block {}
    }

    statement {
      size_constraint_statement {
        comparison_operator = "LT"
        size                = 50

        field_to_match {
          query_string {}
        }

        text_transformation {
          priority = 5
          type     = "NONE"
        }

        text_transformation {
          priority = 2
          type     = "CMD_LINE"
        }
      }
    }

    visibility_config {
      cloudwatch_metrics_enabled = false
      metric_name                = "friendly-rule-metric-name"
      sampled_requests_enabled   = false
    }
  }

  visibility_config {
    cloudwatch_metrics_enabled = false
    metric_name                = "friendly-metric-name"
    sampled_requests_enabled   = false
  }
}
```
### Expected Behavior

I expected the resource `aws_wafv2_rule_group` to just be updated and not recreated when I changed the priority of a rule for example.
### Actual Behavior

When making any changes to the rules, the resource `aws_wafv2_rule_group` is recreated.
```hcl
 - rule { # forces replacement
          - name     = "rule-2" -> null
          - priority = 10 -> null

          - action {

              - block {}
            }

          - statement {

              - size_constraint_statement {
                  - comparison_operator = "LT" -> null
                  - size                = 50 -> null

                  - field_to_match {

                      - query_string {}
                    }

                  - text_transformation {
                      - priority = 2 -> null
                      - type     = "CMD_LINE" -> null
                    }
                  - text_transformation {
                      - priority = 5 -> null
                      - type     = "NONE" -> null
                    }
                }
            }

          - visibility_config {
              - cloudwatch_metrics_enabled = false -> null
              - metric_name                = "friendly-rule-metric-name" -> null
              - sampled_requests_enabled   = false -> null
            }
        }
 + rule { # forces replacement
          + name     = "rule-2"
          + priority = 1

          + action {

              + block {}
            }

          + statement {

              + size_constraint_statement {
                  + comparison_operator = "LT"
                  + size                = 50

                  + field_to_match {

                      + query_string {}
                    }

                  + text_transformation {
                      + priority = 2
                      + type     = "CMD_LINE"
                    }
                  + text_transformation {
                      + priority = 5
                      + type     = "NONE"
                    }
                }
            }

          + visibility_config {
              + cloudwatch_metrics_enabled = false
              + metric_name                = "friendly-rule-metric-name"
              + sampled_requests_enabled   = false
            }
        }

        visibility_config {
            cloudwatch_metrics_enabled = false
            metric_name                = "friendly-metric-name"
            sampled_requests_enabled   = false
        }
    }
```
### Steps to Reproduce

Just change the rule priority

**Comment by bryanlalexander** [comment]
We just ran into this today as well. It does seem overly destructive to delete the resource, in this case, to update the rules.

**Comment by juanpgomez-gsa** [comment]
I am also seeing the same behavior with aws_wafv2_web_acl.

**Comment by hmikerice** [comment]
:+1:

**Comment by AshMenhennett** [comment]
Ran into this recently and look for a resolution - in this case, any removal of a rule from the `aws_wafv2_web_acl` resource results in a tear down of the firewall. This resource is not suitable for a production environment with a **break-glass** scenario that requires updates to the rules in-place to meet Security requirements (I.e. enforce some private access controls).

Happy to put in the steps to reproduce, it's the same scenario as per above, directly targeting the `aws_wafv2_web_acl` resource itself.

**Comment by anGie44** [comment]
A fix has been merged and will release with `v3.2.0` of the Terraform AWS Provider, expected out later today.
